### ISSUE TITLE
Loop Variables

### ISSUE THREAD

**Issue by StephanGarbin** [mitsuba-renderer/drjit#50]
Hello drjit & mitsuba team!

This is an absolutely fantastic release, thank you so much for making this incredible API even better.

Tried looking through the examples and documentation for an answer to the following but couldn't find anything yet, so asking here for a clarification. Let's say I have a loop that does ray marching using path replay backpropagation:

```
        <...>

        loop = mi.Loop(name=f"PRB ({mode.name})",

                       state=lambda: (sampler, ray, L, t, δL, β, active))

        while loop(active):

            p = ray(t)

            <...>
```

Now, say I have two grids (A and B), and I want to use p.z to make a binary decision which of those I'm looking up a value at p in. So, in pseudo-code:
```
if p.z < 0.5:

    <look up values in grid A>

else:

    <look up values in grid B>

```
This decision is fixed, but I do want gradients for the values in grids A and B, In pytorch this could be written using masking, but what is the correct way to write this using drjit / mitsuba's recorded loops?

Many thanks for your time.

Best,
Stephan

**Comment by dvicini** [comment]
Hi Stephan,
DrJit supports both masked assignment and select statements. If I understood your question correctly, you could do something like 
```
value = dr.select(p.z < 0.5, gridA.eval(...), gridB.eval(...))
dr,backward(value)
```
Or you could do 
```
value = gridA.eval(...)
value[p.z >= 0.5] = grid.B.eval(...)
dr.backward(value)
```

Does that answer your question?

Delio

**Comment by StephanGarbin** [comment]
Hi Delio,

Many thanks for the quick response! :) Yep, that answers it perfectly thank you. A quick follow-up: Let's say I'm actually indexing a texture for a specific quantity on the grid, where the texture coordinates are derived from p. In the simplest case `texcoord = [p.x, p.z]`

If I understand correctly, I wouldn't be able to do this, because the state of texcoord would upset the loop construct:

```
        <...>

        loop = mi.Loop(name=f"PRB ({mode.name})", state=lambda: (sampler, ray, L, t, δL, β, active))

        while loop(active):

            p = ray(t)
            texcoord = mi.Vector2f(p.x, p.z)

            <...>
```

What's the recommended way of writing this?

Many thanks again.

Best,
Stephan
